grpc-server/grpc_server.go

```diff
@@ -119,55 +151,71 @@ type storage struct {
 	Filesystems []*filesystem `json:"filesystems,omitempty"`
 }
 
-// exportedHardware transforms hardware that is returned from cacher into what we want to expose to clients
-func exportHardware(hw []byte) ([]byte, error) {
-	exported := &exportedHardwareCacher{}
+func Serve(ctx context.Context, l log.Logger, srv *Server) error {
+	port := env.Int("GRPC_PORT", 42115)
 
-	err := json.Unmarshal(hw, exported)
-	if err != nil {
-		return nil, err
+	serverOpts := make([]grpc.ServerOption, 0)
+
+	// setup tls credentials
+	if *useTLS {
+		creds, err := credentials.NewServerTLSFromFile(*tlsCertPath, *tlsKeyPath)
+		if err != nil {
+			l.Error(err, "failed to initialize server credentials")
+			panic(err)
+		}
+		serverOpts = append(serverOpts, grpc.Creds(creds))
 	}
-	return json.Marshal(exported)
-}
 
-func filterMetadata(hw []byte, filter string) ([]byte, error) {
-	var result bytes.Buffer
-	query, err := gojq.Parse(filter)
+	trustedProxies := xff.ParseTrustedProxies()
+	xffStream, xffUnary := xff.GRPCMiddlewares(l, trustedProxies)
+	streamLogger, unaryLogger := l.GRPCLoggers()
+	serverOpts = append(serverOpts,
+		grpc_middleware.WithUnaryServerChain(
+			xffUnary,
+			unaryLogger,
+			grpc_prometheus.UnaryServerInterceptor,
+		),
+		grpc_middleware.WithStreamServerChain(
+			xffStream,
+			streamLogger,
+			grpc_prometheus.StreamServerInterceptor,
+		),
+	)
+
+	grpcServer := grpc.NewServer(serverOpts...)
+
+	// Register grpc prometheus server
+	grpc_prometheus.Register(grpcServer)
+
+	hegel.RegisterHegelServer(grpcServer, srv)
+
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
 	if err != nil {
-		return nil, err
+		err = errors.Wrap(err, "failed to listen")
+		l.Error(err)
+		panic(err)
 	}
-	input := make(map[string]interface{})
-	err = json.Unmarshal(hw, &input)
+
+	metrics.State.Set(metrics.Ready)
+	//Serving GRPC
+	l.Info("serving grpc")
+	err = grpcServer.Serve(lis)
 	if err != nil {
-		return nil, err
+		l.Fatal(err, "Failed to serve  grpc")
 	}
-	iter := query.Run(input)
-	for {
-		v, ok := iter.Next()
-		if !ok {
-			break
-		}
 
-		if v == nil {
-			continue
-		}
+	return nil
+}
 
-		switch vv := v.(type) {
-		case error:
-			return nil, errors.Wrap(vv, "error while filtering with gojq")
-		case string:
-			result.WriteString(vv)
-		default:
-			marshalled, err := json.Marshal(vv)
-			if err != nil {
-				return nil, errors.Wrap(err, "error marshalling jq result")
-			}
-			result.Write(marshalled)
-		}
-		result.WriteRune('\n')
-	}
+// exportedHardware transforms hardware that is returned from cacher into what we want to expose to clients
+func ExportHardware(hw []byte) ([]byte, error) {
```

Comment:
This doesn't seem like it belong here, maybe in hardware-getter renamed hardware?